### PR TITLE
g79 integration testing after apply catbuffer for elephant

### LINKED ISSUE #79: Integration testing after applying catbuffer
Revisit E2E testing after #73 Apply Catbuffer.
- Fix Standalone and Aggregate Transfer Transaction

### PR DESCRIPTION
**Pull request by fullcircle23**
Fixes for integration and unit tests after applying catbuffer for elephant.
The following 16 E2E tests passed:
1. standaloneTransferTransaction
2. aggregateTransferTransaction
3. standaloneRootRegisterNamespaceTransaction
4. aggregateRootRegisterNamespaceTransaction
5. standaloneSubNamespaceRegisterNamespaceTransaction
6. aggregateSubNamespaceRegisterNamespaceTransaction
7. standaloneMosaicDefinitionTransaction
8. aggregateMosaicDefinitionTransaction
9. standaloneMosaicSupplyChangeTransaction
10. aggregateMosaicSupplyChangeTransaction
11. standaloneLockFundsTransaction
12. aggregateLockFundsTransaction
13. standaloneSecretLockTransaction
14. aggregateSecretLockTransaction
15. standaloneSecretProofTransaction
16. aggregateSecretProofTransaction

Closes #79

### CODE REVIEW

File: src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java [hunk 1]
```diff
@@ -68,41 +71,27 @@
...
-                "-1999805136990834023")); // This namespace is created in functional testing
-    private String namespaceName = "nem2-tests";
-    private MosaicId mosaicId =
-        new MosaicId(
-            new BigInteger("4532189107927582222")); // This mosaic is created in functional testing
+    private NamespaceId rootNamespaceId;
+    private MosaicId mosaicId;
     private Listener listener;
     private String generationHash;
+    private long timeoutSeconds = 20;
```
Comment (rg911) [inline src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java]: Might need longer timeout value.  Probably will be fine on local server, but remote server could take longer. Perhaps we can make it configurable?

File: src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java [hunk 2]
```diff
@@ -364,120 +381,17 @@ void aggregateMosaicSupplyChangeTransaction() throws ExecutionException, Interru
...
-        transactionHttp.announce(signedTransaction).toFuture().get();
+        TransactionAnnounceResponse transactionAnnounceResponse =
+            transactionHttp.announce(signedTransaction).toFuture().get();
+        System.out.println(transactionAnnounceResponse.getMessage());
 
         this.validateTransactionAnnounceCorrectly(
             this.account.getAddress(), signedTransaction.getHash());
     }
 
-    @Test
```
Comment (rg911) [inline src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java]: why are we removing modifyMultisig tx here?

File: src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java [hunk 3]
```diff
@@ -120,15 +120,12 @@
 
     @Test
     void mosaicIdGeneratesCorrectWellKnownId() {
-        int[] nonce = new int[]{0x78, 0xE3, 0x6F, 0xB7};
+        int[] nonce = new int[]{0xB7, 0x6F, 0xE3, 0x78};
```
Comment (rg911) [inline src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java]: where does this hex come from?

File: src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java [hunk 4]
```diff
@@ -144,7 +141,7 @@ void mosaicIdGeneratesCorrectWellKnownId() {
     void mosaicIdGeneratesCorrectGivenNonceAndPublicKey(
```
Comment (rg911) [inline src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java]: Have we tested the idGenerator creates the same Mosaic & Namespace Ids as the other SDKs?

File: src/main/java/io/nem/sdk/model/transaction/UInt64.java [hunk 5]
```diff
@@ -66,6 +67,11 @@ public static BigInteger fromLowerAndHigher(long lower, long higher) {
         return UInt64.fromIntArray(array);
     }
 
+    public static BigInteger fromLowerAndHigher(Number lower, Number higher) {
```
Comment (rg911) [inline src/main/java/io/nem/sdk/model/transaction/UInt64.java]: method doc?

File: src/main/java/io/nem/sdk/model/transaction/IdGenerator.java [hunk 6]
```diff
@@ -21,22 +21,31 @@
...
+
+    public static BigInteger generateMosaicId(final byte[] nonce, final byte[] publicKey) {
+        final byte[] reverseNonce = ByteUtils.reverseCopy(nonce);
+        final byte[] hash = IdGenerator.getHashInLittleEndian(reverseNonce, publicKey);
+        // Unset the high bit for mosaic id
+        return BigInteger.valueOf(ByteBuffer.wrap(hash).getLong() & ~ID_GENERATOR_FLAG);
     }
 
-    public static BigInteger generateNamespaceId(String namespaceName, BigInteger parentId) {
+/*    public static BigInteger generateNamespaceId(String namespaceName, BigInteger parentId) {
```
Comment (rg911) [inline src/main/java/io/nem/sdk/model/transaction/IdGenerator.java]: clean up the comments

File: src/main/java/io/nem/sdk/model/transaction/IdGenerator.java [hunk 7]
```diff
@@ -50,6 +59,17 @@ public static BigInteger generateNamespaceId(String namespaceName, BigInteger pa
         // Set the high bit for namespace id
         hash = (new BigInteger(hash)).or(new BigInteger("8000000000000000", 16)).toByteArray();
         return new BigInteger(hash);
+    }*/
+
+    public static BigInteger generateNamespaceId(final String namespaceName, final BigInteger parentId) {
```
Comment (rg911) [inline src/main/java/io/nem/sdk/model/transaction/IdGenerator.java]: method doc?

File: src/main/java/io/nem/sdk/model/transaction/IdGenerator.java [hunk 8]
```diff
@@ -21,22 +21,31 @@
...
 import java.util.Arrays;
 import java.util.List;
 import java.util.regex.Pattern;
 import org.apache.commons.lang.ArrayUtils;
 
 public class IdGenerator {
+    private static final long ID_GENERATOR_FLAG = 0x8000000000000000L;
 
-    public static BigInteger generateMosaicId(byte[] nonce, byte[] publicKey) {
+/*    public static BigInteger generateMosaicId(byte[] nonce, byte[] publicKey) {
```
Comment (rg911) [inline src/main/java/io/nem/sdk/model/transaction/IdGenerator.java]: Comments

File: src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java (hunk 1 above)
Comment (fullcircle23) [reply src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java]: Added timeoutSeconds to config file.

File: src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java (hunk 2 above)
Comment (fullcircle23) [reply src/integration-test/java/io/nem/sdk/infrastructure/E2ETest.java]: This is part of the aggregate bonded transactions tests which will be included in the next batch of tests and part of a separate ticket as mentioned in #79. I've uncommented it back but note that this and the cosignatureTransaction tests are not working yet.

File: src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java (hunk 3 above)
Comment (fullcircle23) [reply src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java]: This is from the first set of test data in provider() i.e. "B76FE378". The bytes had been reversed for big endian/little endian but don't need to reverse the bytes as this is now taken care within IdGenerator.generateMosaicId(nonce, publicKey). Related fix in test mosaicIdGeneratesCorrectGivenNonceAndPublicKey(...).

File: src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java (hunk 4 above)
Comment (fullcircle23) [reply src/test/java/io/nem/sdk/model/transaction/IdGeneratorTest.java]: Yes. The test data/vectors were copied from ts/js sdk library.

File: src/main/java/io/nem/sdk/model/transaction/UInt64.java (hunk 5 above)
Comment (fullcircle23) [reply src/main/java/io/nem/sdk/model/transaction/UInt64.java]: Added java docs.

File: src/main/java/io/nem/sdk/model/transaction/IdGenerator.java (hunk 6 above)
Comment (fullcircle23) [reply src/main/java/io/nem/sdk/model/transaction/IdGenerator.java]: Done.

File: src/main/java/io/nem/sdk/model/transaction/IdGenerator.java (hunk 7 above)
Comment (fullcircle23) [reply src/main/java/io/nem/sdk/model/transaction/IdGenerator.java]: Done.

File: src/main/java/io/nem/sdk/model/transaction/IdGenerator.java (hunk 8 above)
Comment (fullcircle23) [reply src/main/java/io/nem/sdk/model/transaction/IdGenerator.java]: Done.

**Pull request merged**